### PR TITLE
Feature/test sorting

### PR DESCRIPTION
**Pull request by sajke-shevchenko**

### CODE REVIEW

File: tests/Feature/SampleFeatureTest.php [hunk 1]
```diff
@@ -31,4 +38,288 @@ public function testWithMock()
...
+            return $serviceMock;
+        });
+
+        $count = $resultData->count();
+
+        $this->assertSortingWorks("api/test-order-by", $count, [$sortingField], []);
+    }
+
+    /** Data for testing sorting by single field */
+    public function sortingData()
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:72]: Better give self-descriptive names to all methods, including dataProviders for data-driven tests. Ex. `sortBySingleFieldDataExamples()`

File: test-app/app/Services/MyService.php [hunk 2]
```diff
@@ -2,10 +2,12 @@
...
 
+use Illuminate\Database\Eloquent\Collection;
+
 class MyService
 {
-    public function getData($orderBy): array
+    public function getData($orderBy): Collection
     {
-        return [];
+        return new Collection([]);
```
Comment (populov) [inline test-app/app/Services/MyService.php:11]: Not an issue, but for test it's enough to return array - it is simpler. Were there any issues with it?

File: tests/Feature/SampleFeatureTest.php [hunk 3]
```diff
@@ -31,4 +38,288 @@ public function testWithMock()
...
+    /**
+     * Test sorting by single column.
+     *
+     * @param Collection $resultData Collection of data
+     * @param string $sortingField Sorting field
+     * @param \Exception|null $expectedException Expected exception
+     *
+     * @dataProvider sortingData
+     */
+    public function testSingleSorting(Collection $resultData, string $sortingField, $expectedException = null)
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:51]: I'd better call it `testSortBySingleFieldCheck()` (and, correspondingly `testSortByMultipleFieldsScheck()`)

File: tests/Feature/SampleFeatureTest.php [hunk 4]
```diff
@@ -31,4 +38,288 @@ public function testWithMock()
...
+            return $serviceMock;
+        });
+
+        $count = $resultData->count();
+
+        $this->assertMultiSortingWorks("api/test-order-by", $count, $sortingFields, []);
+    }
+
+    /** Data for testing sorting by several fields */
+    public function multiSortingData()
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:219]: let's rename it accordingly to `sortByMultipleFieldsExamples()`

File: tests/Feature/SampleFeatureTest.php [hunk 5]
```diff
@@ -2,23 +2,30 @@
 
 namespace Saritasa\LaravelTestbed\Tests\Feature;
 
+use Illuminate\Database\Eloquent\Collection;
 use Mockery;
+use PHPUnit\Framework\AssertionFailedError;
+use Saritasa\LaravelTestbed\Traits\ApiListSortingCheck;
 use TestApp\Services\MyService;
 
 class SampleFeatureTest extends TestCase
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:11]: That's definitely not a "SampleFeatureTest" but a true "ApiListSortingCheckTest". A very common practice to name test as "TestedClass" + "Test".

File: tests/Feature/SampleFeatureTest.php [hunk 6]
```diff
@@ -2,23 +2,30 @@
...
 {
+    use ApiListSortingCheck;
+
+    /** Test to check endpoint works */
     public function testHttpRequest()
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:16]: You can either remove this method and `testWithMock()` or remain them in a "SampleFeatureTest" class.

File: tests/Feature/SampleFeatureTest.php [hunk 7]
```diff
@@ -31,4 +38,288 @@ public function testWithMock()
...
+                    ['id' => 2, 'contact' => ['name' => 'Bill']],
+                    ['id' => 3, 'contact' => ['name' => 'Chuck']],
+                    ['id' => 4, 'contact' => ['name' => 'Tim']],
+                    ['id' => 5, 'contact' => ['name' => 'Wood']],
+                    ['id' => 1, 'contact' => ['name' => 'Alex']],
+                ]),
+                'contacts.name',
+                AssertionFailedError::class
+            ],
+            'correct sorted list with nested keys and nullable values (by contact.name)' => [
```
Comment (populov) [inline tests/Feature/SampleFeatureTest.php:138]: Awesome descriptions for test cases! :+1: You definitely got the idea.

**Review by populov — CHANGES_REQUESTED** [review]
Overall is very good, but let's rename few things.

File: tests/Feature/SampleFeatureTest.php (hunk 1 above)
Comment (sajke-shevchenko) [reply tests/Feature/SampleFeatureTest.php]: done.

File: test-app/app/Services/MyService.php (hunk 2 above)
Comment (sajke-shevchenko) [reply test-app/app/Services/MyService.php]: Agree with you.

File: tests/Feature/SampleFeatureTest.php (hunk 3 above)
Comment (sajke-shevchenko) [reply tests/Feature/SampleFeatureTest.php]: done.

File: tests/Feature/SampleFeatureTest.php (hunk 4 above)
Comment (sajke-shevchenko) [reply tests/Feature/SampleFeatureTest.php]: done.

File: tests/Feature/SampleFeatureTest.php (hunk 5 above)
Comment (sajke-shevchenko) [reply tests/Feature/SampleFeatureTest.php:8]: done.

File: tests/Feature/SampleFeatureTest.php (hunk 6 above)
Comment (sajke-shevchenko) [reply tests/Feature/SampleFeatureTest.php:11]: done.

**Pull request merged**